File: java/src/main/java/com/penapereira/patterns/nullobject/OutputLogger.java
package com.penapereira.patterns.nullobject;

import com.penapereira.patterns.runtime.Output;

/** The real object: writes every message to the output. */
public final class OutputLogger implements Logger {

    private final Output out;

    public OutputLogger(Output out) {
        this.out = out;
    }

    @Override
    public void log(String message) {
        out.line("    [log] " + message);
    }
}

File: python/src/patterns/nullobject/output_logger.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from patterns.runtime.contract import Output


class OutputLogger:
    """The real object: writes every message to the output."""

    def __init__(self, out: Output) -> None:
        self._out = out

    def log(self, message: str) -> None:
        self._out.line(f"    [log] {message}")
